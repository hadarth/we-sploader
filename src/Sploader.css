.svg_container {
  position: relative;
  width: 200px;
  height: 200px;
}

.base {
  fill:none;
  stroke:#f3f3f3;
  stroke-width:10;
}

.blue {
  fill:none;
  stroke:#0071bc;
  stroke-width:10;
  stroke-linecap:round;
  stroke-dasharray: 100;
}

.spinit {
  animation: spin 2s linear infinite;
}

@keyframes spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

.loading_value {
  position: absolute;
  font-size: 3em;
}

.container {
  display: flex;
  flex: 1;
  justify-content: center;
  align-items: center;
}