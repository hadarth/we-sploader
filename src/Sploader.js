import React, { Component } from 'react'
import PropTypes from 'prop-types'
import './Sploader.css'
import Button from './Button'
import SploaderGraphics from './SploaderGraphics'



export default class Sploader extends Component {

  constructor(props, context) {
    super(props);
    this.SploaderSvg = React.createRef();
  }


  start = () => this.SploaderSvg.current.style.animationPlayState = ""

  stop = () => this.SploaderSvg.current.style.animationPlayState = "paused"

  render() {
    const { percent } = this.props
    return (
      <div className="container">
        <div className="sploader_container">
          <div className="svg_container">
            <SploaderGraphics percent={percent} SploaderSvg={this.SploaderSvg} />
          </div>
          <div className="loading_value">
            { percent }
            <span className="percent">%</span>
          </div>
        </div>
        <div className="buttons">
          <Button text='start spinning' onClick={this.start}/>
          <Button text='stop spinning' onClick={this.stop}/>
        </div>
      </div>
    );
  }
}

Sploader.propTypes = {
  percent: PropTypes.number
}

Sploader.defaultProps = {
  percent: 15
};