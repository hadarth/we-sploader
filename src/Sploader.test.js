import React from 'react';
import Sploader from './Sploader';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

import enzymeSerializer from 'enzyme-to-json/serializer';
expect.addSnapshotSerializer(enzymeSerializer);

describe("sploader", () => {
  test('should rotate with 13%', () => {
    const wrapper = shallow(<Sploader percent={13}/>)
    expect(wrapper).toMatchSnapshot()
  })

  test('shouuld pass the props', () => {
    const wrapper = shallow(<Sploader percent={20}/>)
    expect(wrapper.instance().props).toMatchSnapshot()
  })

  test('shouuld render with default prop', () => {
    const wrapper = shallow(<Sploader />)
    expect(wrapper.instance().props).toMatchSnapshot()
  })
});
