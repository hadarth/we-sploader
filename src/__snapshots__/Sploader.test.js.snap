// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sploader should rotate with 13% 1`] = `
<div
  className="container"
>
  <div
    className="sploader_container"
  >
    <div
      className="svg_container"
    >
      <SploaderGraphics
        SploaderSvg={
          Object {
            "current": null,
          }
        }
        percent={13}
      />
    </div>
    <div
      className="loading_value"
    >
      13
      <span
        className="percent"
      >
        %
      </span>
    </div>
  </div>
  <div
    className="buttons"
  >
    <Button
      onClick={[Function]}
      text="start spinning"
    />
    <Button
      onClick={[Function]}
      text="stop spinning"
    />
  </div>
</div>
`;

exports[`sploader shouuld pass the props 1`] = `
Object {
  "percent": 20,
}
`;

exports[`sploader shouuld render with default prop 1`] = `
Object {
  "percent": 15,
}
`;
